asciidoc/session-API.adoc

```diff
@@ -0,0 +1,682 @@
+This is carried out by supplying configuration inside the session constructor.
+See <<driver-session-configuration>> for more details.
+
+[[driver-rx-autocommit-transactions]]
+=== Auto-commit transactions
+
+An auto-commit transaction is a basic but limited form of transaction.
+Such a transaction consists of only one Cypher query and is not automatically replayed on failure.
+Therefore any error scenarios will need to be handled by the client application itself.
+
+Auto-commit transactions are intended to be used for simple use cases such as when learning Cypher or writing one-off scripts.
+It is not recommended to use auto-commit transactions in production environments.
+
+[NOTE]
+The only way to execute `PERIODIC COMMIT` Cypher queries is to auto-commit the transaction.
+Unlike other kinds of Cypher query, `PERIODIC COMMIT` queries do not participate in the causal chain.
+Please refer to the <<cypher-manual#query-using-periodic-commit-hint, Cypher Manual -> `PERIODIC COMMIT` query hint>>.
+
+[.tabbed-example]
+.Auto-commit transactions
+========
+[.include-with-dotnet]
+======
+
+.Auto-commit transactions
+[source, csharp]
+----
+include::{dotnet-examples}/ExamplesRx.cs[tags=rx-autocommit-transaction]
+----
+======
+
+[.include-with-java]
+======
+
+.Auto-commit transactions
+[source, java]
+----
+include::{java-examples}/RxTransactionFunctionExample.java[tags=rx-autocommit-transaction]
+----
+======
+
+
+[.include-with-javascript]
+======
+
+.Auto-commit transactions
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=rx-autocommit-transaction]
+----
+======
+========
+
+[[driver-rx-result-consume]]
+=== Consuming results
+
+To consume data from a query in a reactive session, a subscriber is required to handle the results that are being returned by the publisher.
+
+Each transaction corresponds to a data flow which supplies the data from the server.
+Result processing begins when records are pulled from this flow.
+Only one subscriber may pull data from a given flow.
+
+[.tabbed-example]
+.Consuming results
+========
+[.include-with-dotnet]
+======
+
+.Consuming results
+[source, csharp]
+----
+include::{dotnet-examples}/ExamplesRx.cs[tags=rx-result-consume]
+----
+======
+
+[.include-with-java]
+======
+
+.Consuming results
+[source, java]
+----
+include::{java-examples}/RxTransactionFunctionExample.java[tags=rx-result-consume]
+----
+======
+
+
+[.include-with-javascript]
+======
+
+.Consuming results
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=rx-result-consume]
+----
+======
+========
+
+
+=== Cancellation
+
+As per the reactive stream specification, a reactive data flow can be cancelled part way through.
+This prematurely commits or rolls back the transaction and stops the query engine from producing any more records.
+
+
+[[driver-session-configuration]]
+== Session configuration
+
+This section details the configuration options available when constructing a session.
```

Comment:
I propose this formatting to give this paragraph some more space:

[.compact]
`Bookmarks`::
The mechanism ...

`DefaultAccessMode`::
A fallback...

etc.

Reply:
Very nice! Fixed!